yente/data/dataset.py

```diff
@@ -12,13 +13,16 @@
 
 from yente.logs import get_logger
 from yente.data.util import get_url_local_path
+from yente.data.loader import load_json_url
+from yente import settings
 
 log = get_logger(__name__)
 BOOT_TIME = datetime_iso(datetime.utcnow())
 
 
 class Dataset(NKDataset):
     def __init__(self, catalog: DataCatalog["Dataset"], data: Dict[str, Any]):
+        self._available_versions_map: Dict[str, str] = {}
```

Comment:
I am almost wondering if this wants to be a separate object - a delta ops iterator (maybe a DatasetDelta?) that knows the dataset and the version in the index and then does all the handling?

Reply:
It started out like that actually, but I moved away from it as complexity grew. I'll give it another spin and see if it makes sense though :)